mozetl/search/aggregates.py

```diff
@@ -192,16 +194,28 @@ def add_derived_columns(exploded_search_counts):
     '''
     udf_get_search_addon_version = udf(get_search_addon_version, StringType())
 
-    return (
-        exploded_search_counts
-        .withColumn(
-            'type',
-            when(col('source').startswith('sap:'), 'tagged-sap')
+    def _generate_when_expr(sources):
+        if not sources:
+            return 'unknown'
+        return when(col('source').endswith(sources[0]), 'sap').otherwise(
+            _generate_when_expr(sources[1:])
+        )
+    when_expr = when(col('source').startswith('sap:'), 'tagged-sap') \
+        .otherwise(
+            when(col('source').startswith('follow-on:'), 'tagged-follow-on')
```

Comment:
I'm not sure these new formats are correct. The new sources will be:
* `in-content:sap:<code>` -> `tagged-sap`
* `in-content:sap-follow-on:<code>` -> `tagged-follow-on`
* `in-content:organic:<code>` -> `organic`

The new fields are going to look like: `<provider>.in-content:[sap|sap-follow-on|organic]:[code|none]` ([source](https://searchfox.org/mozilla-central/source/toolkit/components/telemetry/Histograms.json#7830)). `main_summary` will parse the source as :`in-content:[sap|sap-follow-on|organic]`